http/http_errors.ts

```diff
@@ -132,6 +138,9 @@ function createHttpErrorConstructor(status: ErrorStatus): typeof HttpError {
       if (options?.expose === false) {
         this.#expose = false;
       }
+      if (options?.headers) {
+        this.#headers = options.headers;
```

Comment:
```suggestion
        this.#headers = new Headers(options.headers);
```

1. with changing it to `HeadersInit` we need to change how we get to the headers.
2. we should avoid "leaking" objects into an error object, by creating a new `Headers` instance we avoid that.